src-java/testing/functional-tests/src/test/groovy/org/openkilda/functionaltests/spec/switches/LagPortSpec.groovy

```diff
@@ -0,0 +1,498 @@
+                        errorMsg   : "Physical port number %d intersects with existing ISLs"
+                ],
+                [
+                        description: "is more than lagOffset",
+                        portNumber : { 2008 },
+                        errorMsg   : "Physical port number %d can't be greater than LAG port offset $lagOffset."
+                ]
+        ]
+    }
+
+    @Tidy
+    def "Unable to create two LAG ports with the same physical ports inside at the same time"() {
+        given: "A switch with a LAG port"
+        def sw = topology.getActiveSwitches().first()
+        def portsArray = topology.getAllowedPortsForSwitch(sw)[-2, -1]
+        def payload = new LagPostDto(portNumbers: portsArray)
+        def lagPort = northboundV2.createLagLogicalPort(sw.dpId, payload).logicalPortNumber
+
+        when: "Try to create the same LAG port with the same physical ports inside"
+        northboundV2.createLagLogicalPort(sw.dpId, payload)
+
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.statusCode == HttpStatus.BAD_REQUEST
+        def errorDetails = exc.responseBodyAsString.to(MessageError)
+        errorDetails.errorMessage ==
+                "Physical ports $payload.portNumbers on switch $sw.dpId already occupied by other LAG group(s)."
+        errorDetails.errorDescription == "Error during LAG create"
+
+        cleanup:
+        lagPort && northboundV2.deleteLagLogicalPort(sw.dpId, lagPort)
+    }
+
+    @Tidy
+    def "Unable to proceed incorrect delete LAG port request (#data.description)"() {
+        when: "Send invalid delete LAG port request"
+        getNorthboundV2().deleteLagLogicalPort(data.swIdForRequest(), data.logicalPortNumber)
+
+        then: "Human readable error is returned"
+        def exc = thrown(HttpClientErrorException)
+        exc.statusCode == HttpStatus.NOT_FOUND
+        def errorDetails = exc.responseBodyAsString.to(MessageError)
+        errorDetails.errorMessage == String.format(data.errorMsg, data.logicalPortNumber, data.swIdForRequest())
+        errorDetails.errorDescription == "Error during LAG delete"
+
+        where:
+        data << [
+                [
+                        description      : "non-existent LAG port",
+                        swIdForRequest   : { getTopology().getActiveSwitches().first().dpId },
+                        logicalPortNumber: 1999, // lagOffset - 1
+                        errorMsg         : "LAG port %d on switch %s not found"
+                ],
+                [
+                        description      : "non-existent LAG port",
+                        swIdForRequest   : { NON_EXISTENT_SWITCH_ID },
+                        logicalPortNumber: 2001, // lagOffset + 1
+                        errorMsg         : "LAG port %d on switch %s not found"
+                ]
+        ]
+    }
+
+
+    //singleSw, singlePortSingleSw on a LAG port ? not sure that need to test
```

Comment:
Should be tested at least manually. Single port traffic will probably have to be tested 100% manually, I doubt we can analyze that via traffgen service

Reply:
test was added